core/src/main/java/org/infinispan/stats/impl/StatsImpl.java

```diff
@@ -47,7 +49,30 @@
    final CacheMgmtInterceptor mgmtInterceptor;
    final Stats source;
 
-   public StatsImpl(AsyncInterceptorChain chain) {
+   public static final class StatsFactory {
```

Comment:
I wonder what it the benefit of having a separate factory class vs just those static create methods added directly to StatsImpl.

Reply:
I do not see the need for a factory here.

Reply:
I said static factory method :) That way throwing an exception would be fine, but if we don't want to throw an exception, we can leave it as a constructor.

Reply:
sorry I created a factory class containing the factory method, in my head I understood that way. I'm still a bit "disturbed" these days in my head